commit 19b94f42dc0afb011a15a03e0659a9d99d2c9062 (refs/changes/27/2427/3)
Author: Andrey Yatskov <andrey.yatskov@silvertreesystems.com>
Date:   2017-08-18T12:55:06-07:00 (2 years, 2 months ago)
    
    ADMINUI-2382 On-Prem- AdminUI possible bug - adding notes to IP's and it automatically makes duplicate entries to other ip's
    Reviewed by: Angela Fong <angela.fong@joyent.com>
    Approved by: Angela Fong <angela.fong@joyent.com>

diff --git a/www/js/components/notes.jsx b/www/js/components/notes.jsx
index 6f518841..1d5f8471 100644
--- a/www/js/components/notes.jsx
+++ b/www/js/components/notes.jsx
@@ -122,11 +122,21 @@ var NotesPanel = React.createClass({
 
 var NotesDropdown = React.createClass({
     getInitialState: function () {
-        return {notes: []};
+        return {
+            notes: [],
+            item: this.props.item
+        };
     },
     propTypes: {
         item: React.PropTypes.string.isRequired
     },
+    componentWillReceiveProps(nextProps) {
+        if (nextProps.item !== this.props.item) {
+            this.setState({item: nextProps.item}, function () {
+                this.fetchNotes();
+            });
+        }
+    },
     handleUnarchive: function (item) {
         var note = new Note(item);
         var req = note.save({archived: false});
@@ -143,7 +153,7 @@ var NotesDropdown = React.createClass({
     },
     handleSave: function (item) {
         var note = new Note({
-            item_uuid: this.props.item,
+            item_uuid: this.state.item,
             note: item.note
         });
         var self = this;
@@ -205,7 +215,7 @@ var NotesDropdown = React.createClass({
         }
         var self = this;
         var collection = new Notes();
-        collection.item_uuid = this.props.item;
+        collection.item_uuid = this.state.item;
         var req = collection.fetch();
         req.done(function (notes) {
             self.setState({notes: notes});
@@ -222,7 +232,7 @@ var NotesDropdown = React.createClass({
         this.setState({dropdown: dropdown});
     },
     render: function () {
-        if (! this.props.item) {
+        if (!this.props.item) {
             console.warn('[Notes] Missing `item` property for notes component, rendering nothing');
             return null;
         }
